tests/src/test_tram.cpp

```diff
@@ -0,0 +1,169 @@
+    std::transform(arr.data(), arr.data() + arr.size(),
+                   finiteBools.mutable_data(), [](dtype x) { return std::isfinite(x); });
+    return std::accumulate(finiteBools.data(), finiteBools.data() + finiteBools.size(), true, std::logical_and<bool>());
+}
+
+
+TEMPLATE_TEST_CASE("TRAM", "[tram]", double, float) {
+
+    GIVEN("Input") {
+        py::scoped_interpreter guard;
+
+        int nThermStates = 3;
+        int nMarkovStates = 5;
+
+        int trajLengths[] = {100, 100, 100};
+
+        auto dtrajs = generateDtrajs(nThermStates, nMarkovStates, trajLengths);
+        auto biasMatrices = generateBiasMatrices<TestType>(nThermStates, dtrajs);
+
+        auto [stateCounts, transitionCounts] = countStates(nThermStates, nMarkovStates, dtrajs);
+
+        auto inputPtr = std::make_shared<deeptime::markov::tram::TRAMInput<TestType>>(
+                std::move(stateCounts), std::move(transitionCounts), dtrajs,
+                biasMatrices);
+
+
+        WHEN("TRAM is constructed") {
+            auto tram = deeptime::markov::tram::TRAM<TestType>(inputPtr, 0);
+
+            THEN("Result matrices are initialized") {
+                REQUIRE(tram.energiesPerThermodynamicState().size() == nThermStates);
+                REQUIRE(tram.biasedConfEnergies().ndim() == 2);
+                REQUIRE(tram.energiesPerMarkovState().size() == nMarkovStates);
+                REQUIRE(tram.biasedConfEnergies().data()[0] == 0);
+            }
+
+            AND_WHEN("estimate() is called") {
+                tram.estimate(1, 1e-8, true);
+
+                TestType LL = tram.computeLogLikelihood();
+
+                THEN("Energies are finite") {
+                    auto thermStateEnergies = tram.energiesPerThermodynamicState();
+                    auto markovStateEnergies = tram.energiesPerMarkovState();
+
+                    REQUIRE(areFinite<TestType>(thermStateEnergies));
+                    REQUIRE(areFinite<TestType>(markovStateEnergies));
+
+                }AND_THEN("Transition matrices are transition matrices") {
+                    auto transitionMatrices = tram.transitionMatrices();
+
+                    REQUIRE(areFinite<TestType>(transitionMatrices));
+                    auto matrixSize = nMarkovStates * nMarkovStates;
+                    auto rowSize = nMarkovStates;
+
+                    for (int K = 0; K < nThermStates; ++K) {
+                        for (int i = 0; i < nMarkovStates; ++i) {
+                            auto rowSum = std::accumulate(transitionMatrices.data() + K * matrixSize + i * rowSize,
+                                                          transitionMatrices.data() + K * matrixSize +
+                                                          (i + 1) * rowSize,
+                                                          0.0, std::plus<TestType>());
+                            REQUIRE(Catch::Detail::Approx(rowSum) == 1.0);
```

Comment:
should be enough to call `Approx(rowSum)`, using a `Detail` namespace doesn't seem right